src/sagemaker/jumpstart/artifacts/model_packages.py

```diff
@@ -96,7 +96,10 @@ def _retrieve_model_package_arn(
         if instance_specific_arn is not None:
             return instance_specific_arn
 
-        if model_specs.hosting_model_package_arns is None:
+        if (
+            model_specs.hosting_model_package_arns is None
+            or model_specs.hosting_model_package_arns == {}
+        ):
             return None
```

Comment:
It's because of the deep merge dict util we have, base fields have `hosting_model_pacakge_arns: None` while config has `hosting_model_package_arns: {"us-west-2": "..."}`. So it's throwing when updating nested dicts. I think the cleanest way is to just initialize base fields with empty dict.